cardano-config/src/Cardano/Config/Types.hs

```diff
@@ -241,9 +243,19 @@ instance FromJSON YamlSocketPath where
   parseJSON invalid = panic $ "Parsing of SocketPath failed due to type mismatch. "
                            <> "Encountered: " <> (T.pack $ show invalid)
 
-parseNodeConfigurationFP :: FilePath -> IO NodeConfiguration
-parseNodeConfigurationFP fp = decodeFileThrow fp
-
+parseNodeConfigurationFP :: ConfigYamlFilePath -> IO NodeConfiguration
+parseNodeConfigurationFP (ConfigYamlFilePath fp) = do
+ nc  <- decodeFileThrow fp
+ let genFile = unGenesisFile $ ncGenesisFile nc
+ exists <- doesFileExist genFile
+ case exists of
+   -- Genesis file is an absolute path
+   True -> pure nc
+   -- Genesis file is a relative path (relative to configuration yaml filepath)
+   False -> do let d = takeDirectory fp
+               pure $ nc { ncGenesisFile = GenesisFile $ d </> genFile }
+
+-- TODO: Make genesisfile relative to configuration file as above.
```

Comment:
I think this is unnecessary. All we need to do is `takeDirectory fp </> genFile`. That will make it relative to the config file's dir if the file is relative, but if it's absolute it'll remain unchanged.

There is no need to do this probing to see if files exist.